### PR TITLE
Create the neighborhood rating crud

### PR DESCRIPTION
**Pull request by lboaventura25**
## Issue 
#7 

## Description
Create the neighborhood rating's CRUD.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/database/models.py [hunk 1]
```diff
@@ -41,4 +42,28 @@ class Occurrence(Base):
...
+class Neighborhood(Base):
+    __tablename__ = 'neighborhood_stay_safe'
+
+    id_neighborhood = Column(Integer, primary_key=True)
+    neighborhood = Column(String, nullable=False)
+    city = Column(String, nullable=False)
+    state = Column(String, nullable=False)
+    rating = relationship("Rating")
+
+
```
Comment (silvasara) [inline src/database/models.py:54]: Put the maximum sizes in the Neighborhood model string fields, as described in the issue

File: src/utils/validators/rating.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    return None
+
+
+def validate_update_rating(body, params):
+    fields = [
+        ('rating_neighborhood', int),
+        ('details', str)
+    ]
+
```
Comment (silvasara) [inline src/utils/validators/rating.py:35]: Adapt update validation in case only one field is being edited

File: src/views/rating.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            if not request.data:
+                return create_response(*type_error)
+            if not all(h in header for h in header_keys):
+                return create_response(*type_error)
+            if not all(header[h] in header_values for h in header_keys):
+                return create_response(*type_error)
+
+        return func(*args, **kwargs)
+
+    return decorated_function
```
Comment (silvasara) [inline src/views/rating.py:31]: Import this decorator from utils/validators/general.py

File: src/views/neighborhood.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                return create_response(*type_error)
+            if not all(h in header for h in header_keys):
+                return create_response(*type_error)
+            if not all(header[h] in header_values for h in header_keys):
+                return create_response(*type_error)
+
+        return func(*args, **kwargs)
+
+    return decorated_function
+
```
Comment (silvasara) [inline src/views/neighborhood.py:32]: Import this decorator from utils/validators/general.py

File: src/views/rating.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+
+@rating_blueprint.route('/rating/<int:id_neighborhood>', methods=['POST'])
```
Comment (silvasara) [inline src/views/rating.py:34]: Protect this endpoint with the decorator validate_token available in utils /validators/general.py and make the necessary changes to the controller functions

File: src/views/rating.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    if request.method == 'GET':
+        response, status = controller.get_one_rating(id_rating)
+
+    elif request.method == 'DELETE':
+        response, status = controller.delete_rating(id_rating)
+
+    elif request.method == 'PATCH':
+        response, status = controller.update_rating(id_rating, request.json)
+
+    return create_response(response, status)
```
Comment (silvasara) [inline src/views/rating.py:64]: Separate the get one method and protect the delete and patch methods with the validate_token decorator available at utils/validators/general.py and make the necessary changes to the controller functions

**Review by silvasara — CHANGES_REQUESTED** [review]

**Review by silvasara — APPROVED** [review]

**Review by renan601 — APPROVED** [review]

**Pull request merged**